include/sys/notify.h

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright (c) 2019 Peter Bigot Consulting, LLC
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_NOTIFY_H_
+#define ZEPHYR_INCLUDE_SYS_NOTIFY_H_
+
+#include <kernel.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup sys_notify_apis Asynchronous Notification APIs
+ * @ingroup kernel_apis
+ * @{
+ */
+
+/* Forward declaration */
+struct sys_notify;
+
+/*
```

Comment:
This looks like a comment that should be in the documentation, so we should put `/**` here.
Same comment to other defines below.
Could we replace these defines by enum as they seem to be sequential and related to each other?

Reply:
It's not public API, but maintainers need to know its intent so it's described.  It's also the value of a bitfield in flags so it shouldn't be an enum.